driver/src/main/java/org/neo4j/driver/internal/metrics/MicrometerConnectionPoolMetrics.java

```diff
@@ -0,0 +1,277 @@
+        Iterable<Tag> tags = Tags.concat( initialTags,
+                                          "address", String.format( "%s:%d", host, address.port() ) );
+
+        Gauge.builder( IN_USE, this::inUse ).tags( tags ).register( registry );
+        Gauge.builder( IDLE, this::idle ).tags( tags ).register( registry );
+        Gauge.builder( CREATING, creating, AtomicInteger::get ).tags( tags ).register( registry );
+        created = Counter.builder( CREATED ).tags( tags ).register( registry );
+        failedToCreate = Counter.builder( FAILED ).tags( tags ).register( registry );
+        closed = Counter.builder( CLOSED ).tags( tags ).register( registry );
+        Gauge.builder( ACQUIRING, acquiring, AtomicInteger::get ).tags( tags ).register( registry );
+        acquired = Counter.builder( ACQUIRED ).tags( tags ).register( registry );
+        timedOutToAcquire = Counter.builder( ACQUISITION_TIMEOUT ).tags( tags ).register( registry );
+        totalAcquisitionTimer = Timer.builder( ACQUISITION ).tags( tags ).register( registry );
+        totalConnectionTimer = Timer.builder( CREATION ).tags( tags ).register( registry );
+        totalInUseTimer = Timer.builder( USAGE ).tags( tags ).register( registry );
+        released = Counter.builder( RELEASED ).tags( tags ).register( registry );
+    }
+
+    @Override
+    public void beforeCreating( ListenerEvent<?> connEvent )
+    {
+        creating.incrementAndGet();
+        connEvent.start();
+    }
+
+    @Override
+    public void afterFailedToCreate()
+    {
+        failedToCreate.increment();
+        creating.decrementAndGet();
+    }
+
+    @Override
+    public void afterCreated( ListenerEvent<?> connEvent )
+    {
+        creating.decrementAndGet();
+        created.increment();
+        Timer.Sample sample = ((MicrometerTimerListenerEvent) connEvent).getSample();
+        sample.stop( totalConnectionTimer );
+    }
+
+    @Override
+    public void afterClosed()
+    {
+        closed.increment();
+    }
+
+    @Override
+    public void beforeAcquiringOrCreating( ListenerEvent<?> acquireEvent )
+    {
+        acquireEvent.start();
+        acquiring.incrementAndGet();
+    }
+
+    @Override
+    public void afterAcquiringOrCreating()
+    {
+        acquiring.decrementAndGet();
+    }
+
+    @Override
+    public void afterAcquiredOrCreated( ListenerEvent<?> acquireEvent )
+    {
+        acquired.increment();
+        Timer.Sample sample = ((MicrometerTimerListenerEvent) acquireEvent).getSample();
+        sample.stop( totalAcquisitionTimer );
+    }
+
+    @Override
+    public void afterTimedOutToAcquireOrCreate()
+    {
+        timedOutToAcquire.increment();
+    }
+
+    @Override
+    public void acquired( ListenerEvent<?> inUseEvent )
+    {
+        inUseEvent.start();
+    }
+
+    @Override
+    public void released( ListenerEvent<?> inUseEvent )
+    {
+        released.increment();
```

Comment:
Likewise, this can be derived from the Timer.

Reply:
Thanks.